pkg/provider/azure_utils.go

```diff
@@ -443,43 +443,49 @@ func getResourceByIPFamily(resource string, isIPv6 bool) string {
 }
 
 // isFIPIPv6 checks if the frontend IP configuration is of IPv6.
-func (az *Cloud) isFIPIPv6(fip *network.FrontendIPConfiguration, pipResourceGroup string, isInternal bool) (isIPv6 bool, err error) {
-	pips, err := az.listPIP(pipResourceGroup)
-	if err != nil {
-		return false, fmt.Errorf("isFIPIPv6: failed to list pip: %w", err)
-	}
+func (az *Cloud) isFIPIPv6(fip *network.FrontendIPConfiguration, isInternal bool) (bool, error) {
 	if isInternal {
 		if fip.FrontendIPConfigurationPropertiesFormat != nil {
 			if fip.FrontendIPConfigurationPropertiesFormat.PrivateIPAddressVersion != "" {
 				return fip.FrontendIPConfigurationPropertiesFormat.PrivateIPAddressVersion == network.IPv6, nil
 			}
-			return net.ParseIP(pointer.StringDeref(fip.FrontendIPConfigurationPropertiesFormat.PrivateIPAddress, "")).To4() == nil, nil
+			if fip.FrontendIPConfigurationPropertiesFormat.PrivateIPAddress != nil {
+				return net.ParseIP(*fip.FrontendIPConfigurationPropertiesFormat.PrivateIPAddress).To4() == nil, nil
+			}
 		}
-		klog.Errorf("Checking IP Family of frontend IP configuration %q of internal Service but its"+
-			" FrontendIPConfigurationPropertiesFormat is nil. It's considered to be IPv4",
+		klog.Errorf("Checking IP Family of frontend IP configuration %q of internal Service but "+
+			"it is not clear. It's considered to be IPv4",
 			pointer.StringDeref(fip.Name, ""))
-		return
+		return false, nil
 	}
 	var fipPIPID string
 	if fip.FrontendIPConfigurationPropertiesFormat != nil && fip.FrontendIPConfigurationPropertiesFormat.PublicIPAddress != nil {
 		fipPIPID = pointer.StringDeref(fip.FrontendIPConfigurationPropertiesFormat.PublicIPAddress.ID, "")
 	}
+	pips, err := az.listPIP(az.ResourceGroup)
```

Comment:
pips may not be in `az.ResourceGroup`. That can be specified from annotation.

Reply:
Good catch! Updated